data-prepper-plugins/sqs-source/src/main/java/org/opensearch/dataprepper/plugins/source/sqs/AwsAuthenticationOptions.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ package org.opensearch.dataprepper.plugins.source.sqs;
+
+ import com.fasterxml.jackson.annotation.JsonProperty;
+ import jakarta.validation.constraints.Size;
+ import software.amazon.awssdk.arns.Arn;
+ import software.amazon.awssdk.regions.Region;
+
+ import java.util.Map;
+ import java.util.Optional;
+
+ public class AwsAuthenticationOptions {
+     private static final String AWS_IAM_ROLE = "role";
+     private static final String AWS_IAM = "iam";
+
+     @JsonProperty("region")
+     @Size(min = 1, message = "Region cannot be empty string")
+     private String awsRegion;
+
+     @JsonProperty("sts_role_arn")
+     @Size(min = 20, max = 2048, message = "awsStsRoleArn length should be between 1 and 2048 characters")
+     protected String awsStsRoleArn;
```

Comment:
is there a reason to make these are protected? Will there be a possible child class or something?

Reply:
great catch, there's no reason for awsStsRoleArn to be protected. I think IntelliJ originally changed it for a test. Changing this to private.